### PR TITLE
Use constructor taking a String as fallback

### LINKED ISSUE #26: ObjectWithFieldsConverter should support converting with com.google.inject.spi.TypeConverter
Currently `ObjectWithFieldsConverter` always relies on a either a registered `ConfigurationConverter` (Plexus API) to set a field value directly or it alternatively creates a new object of the requested type (default constructor leveraged) and tries to set its value via reflection (set or add method).

It should also support `com.google.inject.spi.TypeConverter` for performing the conversion.

### PR DESCRIPTION
**Pull request by kwin**
This closes #26

### CODE REVIEW

File: org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/component/configurator/converters/composite/ObjectWithFieldsConverter.java [hunk 1]
```diff
@@ -51,14 +53,31 @@ public Object fromConfiguration( final ConverterLookup lookup, final PlexusConfi
...
-                new CompositeBeanHelper( lookup, loader, evaluator, listener ).setDefault( bean, value, configuration );
+                // fallback: try with constructor taking single string
+                if ( configuration.getChildCount() == 0 && value instanceof String )
+                {
+                    bean = fromConstructorTakingString( type, (String) value, e );
+                }
+                else
+                {
+                    throw e;
+                }
```
Comment (kwin) [inline org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/component/configurator/converters/composite/ObjectWithFieldsConverter.java:79]: I would like to use https://google.github.io/guice/api-docs/3.0/javadoc/com/google/inject/spi/TypeConverter.html as well, but I don't know how to access the [Injector](https://google.github.io/guice/api-docs/3.0/javadoc/com/google/inject/Injector.html#getTypeConverterBindings()) from here (as this is independent of Plexus Container which holds the injector). @cstamas or @mcculls any idea here?

Comment (kwin) [reply org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/component/configurator/converters/composite/ObjectWithFieldsConverter.java:79]: The only option coming to my mind is passing a `PlexusContainer` to a new constructor of `AbstractComponentConfigurator` (in addition to the default empty constructor) and then pass that instance in calls to `fromConfiguration` of `ConfigurationConverter`. Additionally the `PlexusContainer` would need to be casted to `DefaultPlexusContainer` to somehow extract the Guice `Injector` from there in `ObjectWithFieldsConverter`.

Comment (kwin) [reply org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/component/configurator/converters/composite/ObjectWithFieldsConverter.java:79]: For now I would leave this out, as I don't have a simple solution and only fallback to the public constructor taking the String. @mcculls I am gonna merge next week if I don't hear any complaints as I would like to base my work for Java 8 date/time converts on this.

File: org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/component/configurator/converters/composite/ObjectWithFieldsConverter.java [hunk 2]
```diff
@@ -72,6 +91,39 @@ public Object fromConfiguration( final ConverterLookup lookup, final PlexusConfi
...
+                {
+                    throw precedingException;
+                }
+                else
+                {
+                    throw e;
+                }
+            }
+        }
+        catch ( InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException | SecurityException e )
```
Comment (mcculls) [inline org.eclipse.sisu.plexus/src/main/java/org/codehaus/plexus/component/configurator/converters/composite/ObjectWithFieldsConverter.java]: you could replace several of these with their common super-type, `ReflectiveOperationException` if you want

**Review by mcculls — APPROVED** [review]
LGTM

**Pull request merged**